src/schemaType/DocumentArrayType.ts

```diff
@@ -96,7 +105,7 @@ class DocumentArrayType extends BaseSchemaType {
 	}
 
 	/** Generate subdocument instances */
-	private *makeSubDocument(record: MvRecord): Generator<Document> {
+	private *makeSubDocument(record: MvRecord): Generator<Document<TSchema, TSchemaDefinition>> {
```

Comment:
You learn something new every time. This is my first time noticing the use of the `*` operator here and had to read up on `Generator`s. I could see this being very helpful for large `DEPOSITS` records.